src/main/java/org/marketdesignresearch/mechlib/core/bidder/Bidder.java

```diff
@@ -59,35 +62,18 @@ default BigDecimal getValue(Bundle bundle, Bundle alreadyWon) {
         return BigDecimal.ZERO.max(getValue(bundle.merge(alreadyWon, true)).subtract(getValue(alreadyWon)));
     }
 
-    /**
-     * Asks the bidder a demand query: What bundles would you choose if you'd have to pay certain prices?
-     *
-     * @param prices             the prices
-     * @param maxNumberOfBundles the maximum number of bundles to report
-     * @param allowNegative      whether or not to also report bundles when the utility of paying the given prices for the bundle would be negative
-     *
-     * If {@link WinnerDetermination.PoolMode#MODE_4} is used, the following parameters define some JOpt parameters linked to the solution pool:
-     * @param relPoolTolerance   the relative pool tolerance: {@link SolveParam#SOLUTION_POOL_MODE_4_RELATIVE_GAP_TOLERANCE}
-     * @param absPoolTolerance   the absolute pool tolerance: {@link SolveParam#SOLUTION_POOL_MODE_4_ABSOLUTE_GAP_TOLERANCE}
-     * @param poolTimeLimit      the pool time limit: {@link SolveParam#SOLUTION_POOL_MODE_4_TIME_LIMIT}
-     * @return A list of best bundles
-     */
-    List<Bundle> getBestBundles(Prices prices, int maxNumberOfBundles, boolean allowNegative, double relPoolTolerance, double absPoolTolerance, double poolTimeLimit);
-
     /**
      * Asks the bidder a demand query, without any pool tolerances or time limit.
      * @see #getBestBundles(Prices, int, boolean, double, double, double)
      */
-    default List<Bundle> getBestBundles(Prices prices, int maxNumberOfBundles, boolean allowNegative) {
-        return getBestBundles(prices, maxNumberOfBundles, allowNegative, 0.0, 0.0, -1);
-    }
+    Set<Bundle> getBestBundles(Prices prices, int maxNumberOfBundles, boolean allowNegative);
 
     /**
      * Asks a bidder a demand query, without any pool tolerances or time limit, not accepting negative utility
      * @see #getBestBundles(Prices, int, boolean)
      */
-    default List<Bundle> getBestBundles(Prices prices, int maxNumberOfBundles) {
-        List<Bundle> results = getBestBundles(prices, maxNumberOfBundles, false);
+    default Set<Bundle> getBestBundles(Prices prices, int maxNumberOfBundles) {
+        Set<Bundle> results = getBestBundles(prices, maxNumberOfBundles, false);
```

Comment:
I changed the return type of the demand query to Set and thus dissalow duplicates in the return. There is no described behaviour such that the List is ordered in any way (e.g. by utility). Therefore I think its safe to change this to Set. 

Anyway my implementation uses always a LinkedHashSet and therefore it is still order preserving.

Reply:
That's fine like this 👍

Reply:
I'm not quite sure about this change. According to some existing tests the order of the returned List/Set should be in decreasing order. So the right thing would be a List that is a Set. But given Javas semantics of equals and hashSet this seems not to be possible. Do you know a UniqueList or something like this which could be used here?

Reply:
Ok. I changed this to a LinkedHashSet